busybox: replace nslookup_lede with upstream implementation

PolynomialDivision opened the pull request:
**busybox: remove nslookup_lede/openwrt.patch**

The nslookup_lede/openwrt applet was introduced in de5b8e5. It was introduced because:

    Add a new LEDE nslookup applet which is compatible with musl libc
    and providing more features like ability to specify query type.

    In contrast to busybox' builtin nslookup applet, this variant does
    not rely on libc resolver internals but uses explicit send logic
    and the libresolv primitives to parse received DNS responses.

In busybox this applet is added in https://github.com/mirror/busybox/commit/0dd3be8c0933d5649152b6019b975876e804e22a. In particular, this commit introduces the variable NSLOOKUP_BIG. We set the default to true and so nothing changes.

PolynomialDivision commented:
ping @aparcar

PolynomialDivision commented:
Okay, something is broken:

```
nslookup google.de
**nslookup: can't resolve '(null)': Name does not resolve**

Name:      google.de
Address 1: 142.250.185.67 fra16s48-in-f3.1e100.net
Address 2: 2a00:1450:4001:831::2003 fra24s12-in-x03.1e100.net
```

pmelange commented:
I'm also totally fine with using a non-patched (nslookup) version of busybox, as long as the SRV nslookup feature makes it into 21.02.  But I think it would make sense to include the patch from this PR for now and to work on a new methodology for integrating busybox later. 

It's been quite a long process to finally get this feature, which has been in the works since June 2019.

* 26th June 2019: the Feature request https://bugs.openwrt.org/index.php?do=details&task_id=2341
* 27th June 2019: upstream patch submitted http://lists.busybox.net/pipermail/busybox/2019-June/087359.html
* 30th July 2019: first attempt to include into OpenWrt https://github.com/openwrt/openwrt/pull/2287 with a comment that it needed to be accepted upstream first.
* 27th Oct 2019: patch accepted into busybox https://github.com/mirror/busybox/commit/6b4960155e94076bf25518e4e268a7a5f849308e
* 26th June 2020: busybox 1.32.0 release including SRV support
* 14th Feb 2021: OpenWrt busybox update from 1.31.1 to 1.33.0 https://github.com/openwrt/openwrt/commit/0275ee5dde7c36c925396779dd23d4f470ab40e1
* 3rd May 2021: patch to include SRV for both 21.02 and master #4137 #4138

PolynomialDivision commented on package/utils/busybox/Makefile:
```diff
@@ -93,9 +93,6 @@ LDLIBS:=m crypt
 endif
 
 LDLIBS += $(call BUSYBOX_IF_ENABLED,PAM,pam pam_misc pthread)
-ifeq ($(CONFIG_USE_GLIBC),y)
-  LDLIBS += $(call BUSYBOX_IF_ENABLED,NSLOOKUP_OPENWRT,resolv)
-endif
```
I'm unsure about these lines. Should we also fallback to `!NSLOOKUP_BIG` if we use glibc?

aparcar replied:
I wouldn't rather keep it consistent.

PolynomialDivision commented:
compile and runtested on sxtsq ac 5.

aparcar commented:
I'd first merge https://github.com/openwrt/openwrt/pull/4138 so it can be backported and then update busybox in master. Waiting currently for a comment from @jow-

aparcar commented:
Please rebase.

aparcar commented:
After some testing I pushed the first commit without the $(AUTORELEASE) feature since there were some [concerns](https://github.com/openwrt/openwrt/commit/9ae3c6f94c616cfbf854d3ec749c7fafc9893942) I want to look at first.

I'll test the second patch but want to give the SRV nslookup_lede.c patch a day before backporting.

PolynomialDivision committed:
busybox: remove nslookup_lede/openwrt.patch

The nslookup_lede/openwrt applet was introduced in de5b8e5. It was
introduced because:

  Add a new LEDE nslookup applet which is compatible with musl libc
  and providing more features like ability to specify query type.

  In contrast to busybox' builtin nslookup applet, this variant does
  not rely on libc resolver internals but uses explicit send logic
  and the libresolv primitives to parse received DNS responses.

In busybox this applet is added in 0dd3be8. In particular, this commit
introduces the variable NSLOOKUP_BIG. We set the default to true and
so nothing changes.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
> Please rebase.

I made the appropriate changes. I will runtest again.

PolynomialDivision commented:
> > Please rebase.
> 
> I made the appropriate changes. I will runtest again.

Done.

aparcar commented:
Merged, I couldn't find functional issues and the size is even 1Kb smaller

```
Comparing package sizes...
Change 	Local	Remote 	Package
-1204	230492	231696	busybox
```
